syntax="proto2";

package medical_imaging;

message MsgString{
    required string context = 1;
};

message MsgPoint2 {
    required float x = 1;
    required float y = 2;
};

message MsgPoint2Array {
    required float x = 1;
    required float y = 2;
};

message MsgPaging {
    required int32 page = 1;
};

message MsgMouse {
    required MsgPoint2 pre = 1;
    required MsgPoint2 cur = 2;
    required int32 tag = 3;
};

message MsgCellInfo{
    required int32 id = 1;
    required int32 type = 2;
    required int32 direction = 3;
    required int32 width = 4;
    required int32 height = 5;
};

message MsgInit{
    required int32 pid = 1;
    required string series_uid = 2;
    repeated MsgCellInfo cells = 3; 
};

message MsgResize{
    repeated MsgCellInfo cells = 1;
};

message MsgWorklistItem{
    required string patient_name = 1;
    required string patient_id = 2;
    required string series_uid = 3;
    required string imaging_modality = 4; 
};

message MsgWorklist{
    repeated MsgWorklistItem items = 1;
};
